slave/src/main/java/org/fao/geonet/readonly/InterceptsStartup.java

```diff
@@ -0,0 +1,287 @@
+		}
+	}
+
+	private int harvesterAlreadyCreated(ServerStartup arg0) {
+		HarvesterSettingRepository settingsRepo = arg0.getContext().getBean(HarvesterSettingRepository.class);
+
+		List<HarvesterSetting> harvesters = settingsRepo.findAllChildren(1);
+		for (HarvesterSetting h : harvesters) {
+			for (HarvesterSetting hs : settingsRepo.findAllChildren(h.getId())) {
+				if (hs.getName().equals("site")) {
+					for (HarvesterSetting s : settingsRepo.findAllChildren(hs.getId())) {
+						if (s.getName().equals("name") && s.getValue().equals(key)) {
+							return h.getId();
+						}
+					}
+				}
+			}
+		}
+		return -1;
+	}
+
+	@Transactional(value = TxType.REQUIRES_NEW)
+	private int addMasterHarvester(ServerStartup arg0, Element harvester) {
+		try {
+			HarvestManager hm = arg0.getContext().getBean(HarvestManager.class);
+			return Integer.valueOf(hm.addHarvesterReturnId(harvester, "1"));
+		} catch (Exception e) {
+			log.error("Error initializing MASTER harvester in readonly instance", e);
+		}
+		return -1;
+	}
+
+	private Element createPrivileges() {
+		Element privileges = new Element("privileges");
+		Element group = new Element("group");
+		group.setAttribute("id", "1");
+		addElementWithAttribute(group, "operation", "name", "view");
+		addElementWithAttribute(group, "operation", "name", "dynamic");
+		addElementWithAttribute(group, "operation", "name", "download");
+		privileges.addContent(group);
+		return privileges;
+	}
+
+	private void addElementWithAttribute(Element group, String name, String attribute, String value) {
+		Element operation = new Element(name);
+		operation.setAttribute(attribute, value);
+		group.addContent(operation);
+	}
+
+	private Element createContent() {
+		Element content = new Element("content");
+		addElementWithContentString(content, "validate", "NOVALIDATION");
+		addElementWithContentString(content, "importxslt", "none");
+		return content;
+	}
+
+	private Element createOptions() {
+		Element options = new Element("options");
+		addElementWithContentString(options, "oneRunOnly", "false");
+		addElementWithContentString(options, "overrideUuid", "OVERRIDE");
+		addElementWithContentString(options, "every", frequency);
+		addElementEmpty(options, "status");
+		return options;
+	}
+
+	private Element createSite() {
+		Element site = new Element("site");
+		addElementWithContentString(site, "name", key);
+		addElementWithContentString(site, "host", masterURL);
+		addElementWithContentString(site, "node", masterNode);
+		addElementWithContentString(site, "useChangeDateForUpdate", "true");
+		addElementWithContentString(site, "createRemoteCategory", "true");
+		addElementWithContentString(site, "icon", "undefined");
+		addElementWithContentString(site, "mefFormatFull", "false");
+		addElementEmpty(site, "xslfilter");
+		Element account = new Element("account");
+		addElementWithContentString(account, "use", "false");
+		addElementEmpty(account, "username");
```

Comment:
If you authenticate on Master, then you will have a complete copy. So then if user use API with basic auth, they will get same results in Slave and Master .... but you dropped users. Maybe you could only disabled them ?